common_examples/iperf/main/iperf.c

```diff
@@ -0,0 +1,80 @@
+#include "esp_netif.h"
+#include "esp_eth.h"
+#include "esp_event.h"
+#include "esp_console.h"
+#include "ethernet_init.h"
+#include "iperf_cmd.h"
+#include "sdkconfig.h"
+
+esp_netif_ip_info_t ip_info = {
+    .ip = {.addr = ESP_IP4TOADDR(192, 168, 1, 1)},
+    .netmask = {.addr =  ESP_IP4TOADDR(255, 255, 255, 0)},
+    .gw = {.addr = ESP_IP4TOADDR(192, 168, 1, 1)}
+};
+
+static void start_dhcp_server_at_connection(void *esp_netif, esp_event_base_t base, int32_t event_id, void *data)
+{
+    esp_netif_dhcpc_stop(esp_netif);
+    esp_netif_set_ip_info(esp_netif, &ip_info);
+    esp_netif_dhcps_start(esp_netif);
+}
+
+void app_main(void)
+{
+    uint8_t eth_port_cnt = 0;
+    esp_eth_handle_t *eth_handles;
+    esp_netif_config_t cfg;
+    esp_netif_inherent_config_t eth_behav_cfg;
+    esp_netif_init();
+    esp_event_loop_create_default();
+    ethernet_init_all(&eth_handles, &eth_port_cnt);
+
+#if CONFIG_EXAMPLE_ACT_AS_DHCP_SERVER
+    // Config for ESP32 to act as DHCP server
+    eth_behav_cfg = (esp_netif_inherent_config_t) {
+        .get_ip_event = IP_EVENT_ETH_GOT_IP,
+        .lost_ip_event = 0,
+        .flags = ESP_NETIF_DHCP_SERVER,
+        .ip_info = &ip_info,
+        .if_key = "ETH_DHCPS",
+        .if_desc = "eth",
+        .route_prio = 50
+    };
+#else
+    if (eth_port_cnt == 1) {
+        // Use default config when using one interface
+        //cfg = (esp_netif_config_t) ESP_NETIF_DEFAULT_ETH();
+        eth_behav_cfg = *(ESP_NETIF_BASE_DEFAULT_ETH);
```

Comment:
Consider rename `eth_behav_cfg` to be more clear it is associated with `netif`. Right now, it could be misunderstood as "Ethernet driver behavior".